api/routes/subcriterias/get_sub_criteria.py

```diff
@@ -43,3 +49,160 @@ def return_subcriteria_from_config(sub_criteria_id):
+    @classmethod
+    def get_themes_answers(cls, theme_id: str) -> list[dict]:
+        """function takes a theme_id arg & returns a list
+        of answers with given theme_id.
+        """
+        sub_criterias = get_all_subcriteria()
+        try:
+            return [
+                theme.get("answers")
+                for themes in sub_criterias
+                for theme in themes.get("themes")
+                if theme_id == theme.get("id")
+            ][0]
+        except IndexError:
+            current_app.logger.error(f"Incorrect theme id -> {theme_id}")
+            return f"Incorrect theme id -> {theme_id}"
+
+    @classmethod
+    def get_application_form(cls, app_json_blob):
+        """function return list of all questions from application form"""
+
+        return [
+            questions
+            for forms in app_json_blob["jsonb_blob"]["forms"]
+            for questions in forms["questions"]
+        ]
+
+    @classmethod
+    def convert_boolean_values(cls, themes_answers: list[dict]) -> list[dict]:
+        """function checks boolean values in themes_answers & replace
+        boolean values to string. (False -> "No", True -> "Yes")
+        Args:
+            themes_answers (_type_): array of dict
+        """
+        current_app.logger.info("Converting boolean values to strings")
+        for answers in themes_answers:
+            if "answer" in answers.keys():
+                if answers["answer"] == False:
+                    answers.update(answer="No")
+                if answers["answer"] == True:
+                    answers.update(answer="Yes")
+
+    @classmethod
+    def sort_add_another_component_contents(
+        cls, themes_answers: list[dict]
+    ) -> list[dict]:
+        """function checks for special presentation_type "heading"
+        in array of themes answers, if exists, adds question-value
+        to an answer key for presentation_type "heading" theme and
+        looks for presentation_type of "description" and "amount" with
+        same presentation_type "heading"s field_id, retrieve words only
+        from strings of answers for presentation_type "description" theme
+        & numbers only for presentation_type "amount" theme.
+
+        Args:
+            themes_answers (_type_): array of dict
+        """
+        for heading in themes_answers:
+            try:
+                if heading["presentation_type"] == "heading":
+                    current_app.logger.info(
+                        "mapping add-another component contents"
+                    )
+                    heading["answer"] = heading["question"]
+                    for theme in themes_answers:
+                        if (
+                            heading["field_id"] in theme.values()
+                            and theme["presentation_type"] == "description"
+                        ):
+                            description_answer = [
+                                description.split(":")[0]
```

Comment:
It might be better to split on `" : "` so we don't have unnecessary white-space, also for scenarios where the `description` has a colon (not sure if this can happen) then it would be a tiny bit safer since it would also need the two spaces to split.

Reply:
Actually you can use `rsplit` with a limit so we don't need to worry about description having a color:

(again, don't know if this will ever happen, but better to have peace of mind imo)

```python3
>>> description, amount = 'Test : one : £5053'.rsplit(" : ", 1)
>>> description
'Test : one'
>>> amount
'£5053'
>>>
```

Reply:
That's a good point Thomas! ❤️ 
I have made those changes.